genesyscloud/architect_grammar_language/resource_genesyscloud_architect_grammar_language.go

```diff
@@ -0,0 +1,144 @@
+package architect_grammar_language
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v112/platformclientv2"
+	"log"
+	"strings"
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+)
+
+/*
+The resource_genesyscloud_architect_grammar_language.go contains all the methods that perform the core logic for a resource.
+*/
+
+// getAllAuthArchitectGrammarLanguage retrieves all of the architect grammar languages via Terraform in the Genesys Cloud and is used for the exporter
+func getAllAuthArchitectGrammarLanguage(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := getArchitectGrammarLanguageProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	languages, err := proxy.getAllArchitectGrammarLanguage(ctx)
+	if err != nil {
+		return nil, diag.Errorf("Failed to get grammar languages: %v", err)
+	}
+
+	for _, language := range *languages {
+		languageId := *language.GrammarId + ":" + *language.Language
```

Comment:
Just out of curiosity why are we appending the two strings with a :

Reply:
The language id is in the format `<grammarId>:<languageCode>` and should be returned by the api but there's a bug where the language id is never being returned by the apis so I have to manually create the id by combining the `grammarId` and `languageCode` properties from the api responses. The architect team have are aware of the bug and when it is fixed we can remove these line and just use the `id` property from the api responses.